test/gateway/eip20_cogateway/progress_mint_with_proof.js

```diff
@@ -0,0 +1,530 @@
+    params.storageHash = TestData.gateway.stake.proof_data.storageHash;
+    params.serializedProof = TestData.gateway.stake.proof_data.storageProof[0].serializedProof;
+
+    testEIP20CoGateway = await TestEIP20CoGateway.new(
+      valueToken,
+      testUtilityToken.address,
+      TestData.contracts.anchor,
+      bountyAmount,
+      TestData.contracts.organization,
+      TestData.contracts.gateway,
+      burner,
+    );
+
+    intentHash = coGatewayUtils.hashStakeIntent(
+      params.amount,
+      params.beneficiary,
+      TestData.contracts.gateway,
+    );
+
+    await testUtilityToken.setCoGatewayAddress(testEIP20CoGateway.address);
+
+    messageHash = await testEIP20CoGateway.setMessage.call(
+      intentHash,
+      params.nonce,
+      params.gasPrice,
+      params.gasLimit,
+      params.staker,
+      params.hashLock,
+    );
+
+    await testEIP20CoGateway.setInboxStatus(
+      messageHash,
+      MessageStatusEnum.Declared,
+    );
+
+    await testEIP20CoGateway.setMessage(
+      intentHash,
+      params.nonce,
+      params.gasPrice,
+      params.gasLimit,
+      params.staker,
+      params.hashLock,
+    );
+
+    await testEIP20CoGateway.setMints(
+      messageHash,
+      params.beneficiary,
+      params.amount,
+    );
+
+    await testEIP20CoGateway.setStorageRoot(
+      params.blockNumber,
+      params.storageHash
+    );
+
+  });
+
+  it('should emit event MintProgressed', async function () {
+
+    let estimatedReward = await getMaxReward(TestData);
+    let estimatedStakeAmount = params.amount.sub(estimatedReward);
+
+    let response = await testEIP20CoGateway.progressMintWithProof(
+      messageHash,
+      params.serializedProof,
+      params.blockNumber,
+      MessageStatusEnum.Declared,
+      {from: facilitator},
+    );
+
+    let expectedEvent = {
+      MintProgressed: {
+        _messageHash: messageHash,
+        _staker: params.staker,
+        _stakeAmount: params.amount,
+        _mintedAmount: estimatedStakeAmount,
+        _rewardAmount: estimatedReward,
+        _proofProgress: true,
+        _unlockSecret: Utils.ZERO_BYTES32
+      }
+    };
+
+    let eventData = response.logs;
+    Utils.validateEvents(eventData, expectedEvent);
+
+  });
+
+  it('should return correct params', async function () {
+
+    let response = await testEIP20CoGateway.progressMintWithProof.call(
+      messageHash,
+      TestData.gateway.stake.proof_data.storageProof[0].serializedProof,
+      new BN(TestData.gateway.stake.return_value.block_number, 16),
+      MessageStatusEnum.Declared,
+      {from: facilitator},
+    );
+
+    assert.strictEqual(
+      response.beneficiary_,
+      params.beneficiary,
+      `Beneficiary address is ${response.beneficiary_}. It should be ${params.beneficiary}`,
+    );
+
+    assert.strictEqual(
+      params.amount.eq(response.stakeAmount_),
+      true,
+      `Staked amount ${response.stakeAmount_} from event should be ${params.amount}.`,
```

Comment:
These sentences are wrongly formed. It will print like  This is returned value.
`Stake amount 50 is from should be 60`. 
Also check other sentences in the file.
Also for BN please add toString(10) to print the reason readable.